core/src/main/java/org/kiwix/kiwixmobile/core/data/KiwixRoomDatabase.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2023 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.data
+
+import android.content.Context
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+import org.kiwix.kiwixmobile.core.BuildConfig
+import org.kiwix.kiwixmobile.core.dao.RecentSearchRoomDao
+import org.kiwix.kiwixmobile.core.dao.entities.RecentSearchRoomEntity
+import org.kiwix.kiwixmobile.core.data.remote.FdroidDatabaseCallback
+import org.kiwix.kiwixmobile.core.data.remote.RoomDatabaseCallback
+
+@Suppress("UnnecessaryAbstractClass")
+@Database(entities = [RecentSearchRoomEntity::class], version = 1)
+abstract class KiwixRoomDatabase : RoomDatabase() {
+  abstract fun recentSearchRoomDao(): RecentSearchRoomDao
+
+  companion object {
+    private var db: KiwixRoomDatabase? = null
+    fun getInstance(context: Context): KiwixRoomDatabase {
+      val callback =
+        if (BuildConfig.BUILD_TYPE != "fdroid")
```

Comment:
@gouri-panda This always returns true, you forgot to define the `fdroid variant` in the `core module` which we placed last time in room PR.